source/parser/util.civet

```diff
@@ -123,8 +123,6 @@ function isFunction(node: ASTNode): node is FunctionNode
       type is "FunctionExpression"
       type is "ArrowFunction"
       type is "MethodDefinition"
-      "async" in node and !!node.async
-      // do blocks can be marked async to prevent automatic await
```

Comment:
Interesting - I guess we don't need to mark `async do` blocks as functions after all? Presumably they get wrapped in IIFE by this point.

Reply:
Yeah, I think we're creating proper function nodes everywhere now.